### PR TITLE
modules: Universal tracy macros (SceAudiodecUser as example)

### PR DESCRIPTION
**Pull request by bookmist**
I try to make universal macro, which can be used to log function parameters with tracy. Now this PR is only a demo, with one module as example.

### CODE REVIEW

**Comment by EXtremeExploit** [comment]
hmm having the tracy macros defined in the module is not that great, you would have to copy pase them anywhere, maybe it would he nice to have them in utils or something like that, they will be used around the project

**Comment by bookmist** [comment]
I was going to do so, but had no idea where to put this universal macro. So if utils is a good place - it's great.

File: vita3k/modules/include/modules/tracy.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#pragma once
+
+#include <mem/state.h>
```
Comment (EXtremeExploit) [inline vita3k/modules/include/modules/tracy.h:20]: also `#include <sstream>` for stringstream on linux end so compiler doesn't give an error on some systemd

File: vita3k/modules/include/modules/tracy.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// I'm not sure it's needed only in case of tracy enabled
+template <typename T>
+std::string to_debug_str(const MemState &mem, T data) {
+    std::stringstream datass;
+    datass << data;
+    return datass.str();
+}
+
+template <>
+std::string to_debug_str<std::string>(const MemState &mem, std::string data) {
```
Comment (EXtremeExploit) [inline vita3k/modules/include/modules/tracy.h:31]: make this inline

File: vita3k/modules/include/modules/tracy.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    return data;
+}
+
+std::string to_debug_str(const MemState &mem) {
```
Comment (EXtremeExploit) [inline vita3k/modules/include/modules/tracy.h:35]: this inline too

**Review by EXtremeExploit — CHANGES_REQUESTED** [review]

File: vita3k/modules/include/modules/tracy.h (hunk 3 above)
Comment (bookmist) [reply vita3k/modules/include/modules/tracy.h]: done. Thanks for help

**Review by EXtremeExploit — APPROVED** [review]

**Pull request merged**